legacy/src/main/java/org/opensearch/sql/legacy/query/join/OpenSearchJoinQueryAction.java

```diff
@@ -34,9 +37,14 @@ public SqlElasticRequestBuilder explain() throws SqlParseException {
     JoinRequestBuilder requestBuilder = createSpecificBuilder();
     fillBasicJoinRequestBuilder(requestBuilder);
     fillSpecificRequestBuilder(requestBuilder);
+    updateRequestWithPitId(requestBuilder);
```

Comment:
As I understand, PIT will be created in PIT handler. We need to make sure it won't be created until execution, ex. EXPLAIN shouldn't create a PIT?

Reply:
That make sense. RequestBuilder is filled here and I need pit id in request, that's why I added. Checking what is alternate option, if you have any suggestion on how to get pit id if not here, please comment.

Reply:
Added pit to request before creating executor. Can you please take another look?

Reply:
I still feel we shouldn't create PIT in request builder. Could you check where we created Scroll context previously? I think request builder should just build request for later execution. Probably that's why there is no dependency on `Client` in request builder.

Reply:
SQL plugin rebuild the context on the fly, the scroll context will be handled natively by OpenSearch. 

(Please see **_3.6.3.3 Context Rebuild_** from  https://github.com/opensearch-project/sql/blob/main/docs/dev/opensearch-pagination.md#36-different-approaches-for-different-sql-queries )


For scroll, we only needed timeout for first request and scroll_id (got this from previous response)
for subsequent request. Only one request was enough to get datarows.
But with PIT, we need extra request to create PIT and that pitId from response is required in every search request that's why added in request builder.